### ISSUE TITLE
Discussion: User feedback about moderation

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#699]
@kevincrepin @anettlinno @ilmartyrk 

This is just from one user test, but I think he makes some points that seem quite common sense, so sharing here :) 

Use case was that he was trying out the platform for use as a live discussion and voting tool during an in-person event. There would be a room full of (200 or so) young people, watching 5 MEP candidates speeches. During each speech, the young people would be able to add their reflections on the speech in the discussion. At the end of each speech, they would be able to vote from 1-5 on to what extent they we convinced by their speech. (It's a interesting use case we've not much considered before, but he is keen to use our platform for this). 

On the moderation feature, he pointed out two things: 

1. As admin, you can't directly moderate an argument. You can report it, but if he as admin spots that there is an offensive post, he wouldn't be reporting it, he would be moderating it himself. 

@kevincrepin Do we need an additional option here for admins, that says "Moderate" (either instead or as well as 'Report', not sure - there might be cases where someone with admin rights wants to report it to a fellow admin?), and means they skip the reporting step and go straight to hiding or deleting the content? And if so what would the flow for this be? 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/925bc9bd-deb5-4f54-8095-e97f71d24b73)

2. If someone reports some content, he would want something to pop-up in the topic to notify him that they have. Obviously he could get this from the topic activity feed, but he didn't think to look there.  After the test, he noticed that he got an email telling him that someone had reported an argument, but during the test (which would have been during the event if it had been a real life situation) he didn't see it. (He then wrote to me to say he had now seen the email, so understood now how he would know, but that it should still be in the topic itself). Obviously he could get this info from the activity feed, but he wasn't aware of this (people aren't looking at it much in the tests), and he would have to keep checking every few seconds to see if the notifications are just for arguments being added, etc, or are for content being reported - so not really practical.

@kevincrepin Is there away we can make reported content appear to admins when they are in the topic itself, for such occasions where it is more 'live' oversight of a topic while the action is happening (as is the case for hybrid meetings too), rather than the more asynchronous oversight for which email notifications makes sense?

**Comment by kevincrepin** [comment]
I think we might need to introduce an entirely new section for admins on reported topics and comments and how to manage them. Right now I agree there's the topic activity and then mails, but both are not really ideal for managing or having the overview.

Also we can show in a topic that there are comments that need moderating, similar to when we show a topic has been reported, but this should limit the amount of reports as well. Imagine in a big discussion there are 20 comments flagged. It's not really in the topic that we need to show that this has happened beyond a general "x-amount of comments have been reported. Please have a look and do your job" :)

A separate "report management" page needs more thinking though and is probably something for Phase II. 

In short-term what we can do is reuse the same notification as we have for reported topic and adjust the copy to indicate that there are reported comments.

![Screenshot 2024-01-26 at 08 07 44](https://github.com/citizenos/citizenos-fe/assets/26792281/606feb6a-e970-4d91-992d-299b373110d1)

Adding extra menu item for admin to immediately moderate a comment:
![Screenshot 2024-01-26 at 08 20 29](https://github.com/citizenos/citizenos-fe/assets/26792281/fa9a9a2d-2d63-40fb-923b-e053aa724620)

And a new dialog window with the moderation details:
![Screenshot 2024-01-26 at 08 22 36](https://github.com/citizenos/citizenos-fe/assets/26792281/f12552f7-06b3-409b-b641-6b0e6a67b170)

**Comment by kevincrepin** [comment]
@ilmartyrk if the above is approved then you can get the designs from https://www.figma.com/file/gOJ4i00538e0qaP6IKIPVE/Platform-2.0-UI?type=design&node-id=11577%3A76229&mode=design&t=yoDKb7xLADti0hlu-1

**Comment by BeccaMelhuish** [comment]
Ah, though this flow for topic reporting was something that I saw previously is abit messy/confusing (e.g. see [here](https://github.com/citizenos/citizenos-fe/issues/443#issuecomment-1830296348)), but seeing as for topics its was only for internal use (i.e. one of the team who deals with reported topics) I just left it for Phase II (or later) when we have time to do a review of it and really work it out properly (there were several different issues with the moderation process including emails, so it seems we need a quite thorough review of it). 

I'd need to (or whoever would be doing the moderation process review) get my head back into what the issues were (I know they related to moderation emails as well, and some confusion there), but here are a few thoughts on this that I have for now:

Here below, do we need 3 different things? Could we not just have 1 window - you see the report and the reason, and you decide either to hide (or delete? see comment at end) the content or allow the content. Allowing it could be either because you don't agree with the person who reported it (i.e. it's not actually offensive, in your opinion), or because the person who wrote it has since edited it. (Ideally this edit would show up as well, if so). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/e1441358-3e6e-43c9-92a8-e27c40e48d44)

Also, I'm wondering what happens when there is more than 1 bit of content to check? I.e. in the example, there are 12, but in the pop-ups there is only 1? 

And a few copy points: 

- As they're either arguments or replies, in the emails we decided on 'discussion contributions'. So here, would be 'reported contributions', and same throughout for everywhere it says 'comment'? 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/885f99eb-0914-4d06-b979-5b972fc6fd76)

- Except here, where it could just be removed:
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/c8c6ca14-b66f-4fd9-9686-1da7ceae01c7)

- If we do keep this separate window, would be 'Moderate contribution'?
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/4fac5043-fb9a-4bad-b47d-6690a410bebb)

- Here it should be 'for moderating': 
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/7b0f4c8b-d683-4f89-911c-85cefed9fe06)

Also, I understand we have a legal obligation (hopefully @anettlinno can confirm) to delete things like hate speech (see this [issue here](https://github.com/citizenos/citizenos-fe/issues/253)), but I've never understood how this happens in practice? We don't seem to ever give admins an option to delete rather than hide, or explain which need deleting or hiding? (Maybe @ilmartyrk knows something about this). 

Here are some other issues relating to the need for moderation flow if this helps to work anything out:
- https://github.com/citizenos/citizenos-fe/issues/443
- https://github.com/citizenos/citizenos-fe/issues/336 
- https://github.com/citizenos/citizenos-fe/issues/459
- https://github.com/citizenos/citizenos-fe/issues/483
- https://github.com/citizenos/citizenos-fe/issues/253

**Comment by kevincrepin** [comment]
I kinda feel that there is no "quick fix" for this. I have some ideas on improving moderating in general, but they are more elaborate and require more time. Can't do this before 31st, but could pick it up next week.

**Comment by BeccaMelhuish** [comment]
OK yes I have removed the pre UX test label now then, and hopefully @anettlinno can guide re when we should try to work this out, but I think ideally if we can do it in time to have had at least a couple of tests that look at moderation (this week's ones wouldn't have a lot of time for it anyway) before launch that would be ideal :)

**Comment by BeccaMelhuish** [comment]
Update from discussion with @anettlinno in roadmap meeting: moderation needs a full review and journey-mapping, it's not in Phase I scope so adding Phase II label now :)
